docs/x509/reference.rst

```diff
@@ -1412,24 +1406,19 @@ General Name Classes
 
     .. versionadded:: 0.9
 
-    .. versionchanged:: 2.1
-
-    .. warning::
-
-        Starting with version 2.1 :term:`U-label` input is deprecated. If
-        passing an internationalized domain name (IDN) you should first IDNA
-        encode the value and then pass the result as a string. Accessing
-        ``value`` will return the :term:`A-label` encoded form even if you pass
-        a U-label. This breaks backwards compatibility, but only for
-        internationalized domain names.
+    .. versionchanged:: 3.1
 
+        :term:`U-label` support has been removed. Encode them to
+        :term:`A-label` before use.
 
     This corresponds to an email address. For example, ``user@example.com``.
 
     :param value: The email address. If the address contains an
```

Comment:
We require these to be text (which we encode to bytes via `.encode('ascii')`), but should we loosen these to allow bytes now?

Reply:
Nah, I don't see why.